[[ Bug 22705 ]] Resolve crash focusing on unopen stack

montegoulding opened the pull request:
This patch resolves a crash where `controlgainedfocus` is called with a stack
parameter which is unopened and therefore `p_stack -> getwindow()` returns
`nullptr`.

runrevmark commented:
@livecode-vulcan review ok 29ba633

runrevmark commented:
@livecode-vulcan review not ok 29ba633

runrevmark reviewed (changes requested):
This needs a release note.

livecode-vulcan commented:
💔 review by @runrevmark not ok 29ba6335f28cfe59e6369b21c54873d5c130fc0e

montegoulding committed:
[[ Bug 22705 ]] Resolve crash focusing on unopen stack

This patch resolves a crash where `controlgainedfocus` is called with a stack
parameter which is unopened and therefore `p_stack -> getwindow()` returns
`nullptr`.

runrevmark commented:
@livecode-vulcan review ok e46e53a

livecode-vulcan commented:
💙 review by @runrevmark ok e46e53aaee761ab0241a1eaef2c319aff7a6927a

livecode-vulcan commented:
😎 test success e46e53aaee761ab0241a1eaef2c319aff7a6927a

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/841)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/250)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/250)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/250)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/250)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2367)
* try-community-universal-ios-iphoneos13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.5/builds/18)
* try-community-universal-ios-iphonesimulator13.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.5/builds/18)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2395)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2383)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2382)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2417)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2357)
